manim/mobject/types/vectorized_mobject.py

```diff
@@ -994,22 +1070,111 @@ def consider_points_equals_2d(self, p0: np.ndarray, p1: np.ndarray) -> bool:
 
         Returns
         -------
-        bool
-            whether two points considered close.
+        bool | np.ndarray
+            Whether the points p0 and p1 are considered close or not.
         """
-        rtol = 1.0e-5  # default from np.isclose()
+        p0 = np.asarray(p0)
+        p1 = np.asarray(p1)
         atol = self.tolerance_for_point_equality
-        if abs(p0[0] - p1[0]) > atol + rtol * abs(p1[0]):
+
+        # Case 1: single pair of points
+        if p0.ndim == 1 and p1.ndim == 1:
+            if abs(p0[0] - p1[0]) > atol:
+                return False
+            if abs(p0[1] - p1[1]) > atol:
+                return False
+            return True
+
+        # Case 2: multiple pairs of points
+        # Ensure that we're only working with 2D components
+        p0 = p0.reshape(-1, self.dim)[:, :2]
+        p1 = p1.reshape(-1, self.dim)[:, :2]
+        is_close = abs(p1 - p0) <= atol
+        return is_close[:, 0] & is_close[:, 1]
+
+    def consider_points_different(
```

Comment:
Hmm, I mainly did this to:
- avoid more function calls which add a little bit of overhead
- avoid negating the result of `consider_points_equals`, because it is also slightly faster to directly compute if the two points are different, rather than computing if they're equal, then negating the result

Both methods could be used everywhere: `consider_points_different` is crucial for splitting points into subpaths, and `consider_points_equals` is necessary for methods like `Camera.set_cairo_context_path` which checks if the subpaths should be closed.

As these methods are essential, one would want them to be as fast as possible, and that's why I did that. However, I'm myself also bothered by the duplication of code, and I'm wondering if the small gains in speed are **really** worth it.

Reply:
In the end, I decided to get rid of `VMobject.consider_points_different` and `VMobject.consider_points_different_2d`, and just use the negation of their "equals" counterparts.